esrally/mechanic/launcher.py

```diff
@@ -340,7 +352,7 @@ def _set_env(self, env, k, v, separator=' '):
             if k not in env:
                 env[k] = v
             else:  # merge
-                env[k] = v + separator + env[k]
+                env[k] = env[k] + separator + v
```

Comment:
Can you please explain why you've changed the order here?

Reply:
For some reason JFR telemetry would not start and rally would never return with pid if the the options didn't start with "-XX:+UnlockDiagnosticVMOptions" That's how it was originally, but now that we changed the telemetry to return list instead of string the order got reversed when we were adding them to the env[k]. So I reversed again so it's in the original order. I could change the order in telemetry instead if you prefer.

Reply:
I see. That's very subtle but it makes sense that we preserve the order in the merge routine (i.e. as you've implemented it now). Can you please add a test so we ensure that `-XX:+UnlockDiagnosticVMOptions` is the first option when we activate multiple telemetry devices (you can just use the full list of JVM args in the assertion)?

Reply:
Did a bit more testing, so it seems that order of options for other telemetry devices doesn't matter. It's that specifically for jfr that `-XX:+UnlockDiagnosticVMOptions` has to come before  `-XX:+DebugNonSafepoints` but other telemetry can be before or after jfr options. So maybe modifying order in telemery for jfr makes more sense than in `_set_env`.. WDYT?

Reply:
Yes, your proposal makes sense to me.